### PR TITLE
test(cli): add test for creating and building a project

### LINKED ISSUE #54: Console errors on npm run dev
### Describe the bug

image

### Link to a StackBlitz project which shows the error

_No response_

### Steps to reproduce

npm i
npm run dev

### Expected behavior

no errors

### Screenshots

_No response_

### Platform

Windows, Chrome. Node 20, npm 10.7.0

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SamVerschueren**
fixes #54

The idea here is that CI also creates the template (which gets generated with `npm create tutorial`) and builds it. This way we can verify that we don't break the template if we update dependencies.

The PR also tests on Windows and addresses some issues with globs that didn't pick up the correct files.

### CODE REVIEW

**Review by Nemikolh — APPROVED** [review]
Yay!! Amazing :star_struck: 

This should close #54

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -7,8 +7,21 @@ on:
 jobs:
   test:
```
Comment (d3lm) [inline .github/workflows/ci.yaml:8]: Unrelated but should we rename this since it's not just tests?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -7,8 +7,21 @@ on:
...
     name: Test
```
Comment (d3lm) [inline .github/workflows/ci.yaml:9]: Same for the name. WDYT?

**Comment by Nemikolh** [comment]
@d3lm Hey! Do you think you could update the required status checks to match what's in this PR?

File: packages/cli/tests/create-tutorial.test.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+beforeAll(async () => {
+  await fs.rm(tmpDir, { force: true, recursive: true });
+  await fs.mkdir(tmpDir);
+});
+
+afterAll(async () => {
+  await fs.rm(tmpDir, { force: true, recursive: true });
+});
+
+test('test creating a project', async () => {
```
Comment (d3lm) [inline packages/cli/tests/create-tutorial.test.ts:18]:
```suggestion
test('create a project', async () => {
```

File: packages/cli/tests/create-tutorial.test.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      TK_DIRECTORY: path.resolve(cwd, '..'),
+    },
+  });
+
+  const projectFiles = await fs.readdir(dest, { recursive: true });
+
+  expect(projectFiles.map(normaliseSlash).sort()).toMatchSnapshot();
+});
+
+test('test creating and building a project', async () => {
```
Comment (d3lm) [inline packages/cli/tests/create-tutorial.test.ts:34]:
```suggestion
test('create and build a project', async () => {
```

**Comment by d3lm** [comment]
> Do you think you could update the required status checks to match what's in this PR?

Yep for sure!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (SamVerschueren) [reply .github/workflows/ci.yaml:8]: I still feel it's mainly testing related. Linting (prettier) and checking if the project builds is in my opinion part of testing. But feel free to suggest a different name though.

**Review by Nemikolh — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (d3lm) [reply .github/workflows/ci.yaml:8]: I mean we can keep it as is. From the top of my head I don't have a better name. Personally I would have different jobs and separate some of the stuff but like I said, we can keep it as is.

**Comment by d3lm** [comment]
@SamVerschueren All status checks updated.

**Review by Nemikolh — APPROVED** [review]
Amazing work!! :muscle: :clap:

**Pull request merged**